casper/src/main/resources/Registry.rho

```diff
@@ -51,10 +61,27 @@ in {
   //     }
   //   }
   // }
-
+  contract createSecurityCheck(lib) = {
+    let @pubKey <-
+      // fc743bd08a822d544bfbe05a5663fc325039a44c8f0c7fbea95a85517da5c36b private
+      "04e2eb6b06058d10b30856043c29076e2d2d7c374d2beedded6ecb8d1df585dfa583bd7949085ac6b0761497b0cfd056eb3d0db97efb3940b14c00fff4e53c85bf".hexToBytes()
+    in {
+      contract @(*lib, "securityCheck")(@deployerID, ret) = {
```

Comment:
Hm, we have similar situation with this contract as with TreeHashMap, it's used in Registry so it cannot be defined in its own file. Please use only this definition to register it and remove separate file.

---

What is the purpose of `lib` here? This contract should just be to check (compare) public key defined in config with supplied _deployerId_. So this wrapper contract is not needed.

---

Name should also not include Coop because it can be used by anyone running their network.
Maybe we can call it `configPublicKeyCheck`?

Reply:
If `configPublicKeyCheck` is defined like this.
```scala
contract configPublicKeyCheck(deployerID, ret) = {
  let @pubKey <- 
    // fc743bd08a822d544bfbe05a5663fc325039a44c8f0c7fbea95a85517da5c36b private
    "04e2eb6b06058d10b30856043c29076e2d2d7c374d2beedded6ecb8d1df585dfa583bd7949085ac6b0761497b0cfd056eb3d0db97efb3940b14c00fff4e53c85bf".hexToBytes()
  in {
    new DeployerIdOps(`rho:rchain:deployerId:ops`), deployerPubKeyBytesCh in {
      DeployerIdOps!("pubKeyBytes", deployerID, *deployerPubKeyBytesCh) |
      for (@deployerPubKeyBytes <- deployerPubKeyBytesCh){
        if (deployerPubKeyBytes == pubKey){
          ret!((true, Nil))
        } else {
          ret!((false, "You have no permissions to update system contract!"))
        }
      }
    }
  } 
}
```
Then each system contract should define its own securityCheck method.
```scala
contract @(*<lib>, "securityCheck")(@deployerID, @return) = {
  configPublicKeyCheck(deployerID, return)
}
```